internal/librarian/create.go

```diff
@@ -0,0 +1,114 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package librarian
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/googleapis/librarian/internal/config"
+	"github.com/googleapis/librarian/internal/librarian/internal/rust"
+	"github.com/googleapis/librarian/internal/yaml"
+	"github.com/urfave/cli/v3"
+)
+
+var (
+	errUnsupportedLanguage          = errors.New("library creation is not supported for the specified language")
+	errOutputFlagRequired           = errors.New("output flag is required when default.output is not set in librarian.yaml")
+	errServiceConfigAndSpecRequired = errors.New("both service-config and specification-source flags are required for creating a new library")
+	errMissingNameFlag              = errors.New("name flag is required to create a new library")
+	errNoYaml                       = errors.New("unable to read librarian.yaml")
+)
+
+func createCommand() *cli.Command {
+	return &cli.Command{
+		Name:      "create",
+		Usage:     "create a new client library",
+		UsageText: "librarian create --name <name> --specification-source <path> --service-config <path>",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "name",
+				Usage: "library name",
+			},
+			&cli.StringFlag{
+				Name:  "specification-source",
+				Usage: "path to the specification source (e.g., google/cloud/secretmanager/v1)",
+			},
+			&cli.StringFlag{
+				Name:  "service-config",
+				Usage: "path to the service config",
+			},
+			&cli.StringFlag{
+				Name:  "output",
+				Usage: "output directory (optional, will be derived if not provided)",
+			},
+			&cli.StringFlag{
+				Name:  "specification-format",
+				Usage: "specification format (e.g., protobuf, discovery)",
+				Value: "protobuf",
+			},
+		},
+		Action: func(ctx context.Context, c *cli.Command) error {
+			name := c.String("name")
+			specSource := c.String("specification-source")
+			serviceConfig := c.String("service-config")
+			output := c.String("output")
+			specFormat := c.String("specification-format")
+			if name == "" {
+				return errMissingNameFlag
+			}
+			return runCreate(ctx, name, specSource, serviceConfig, output, specFormat)
+		},
+	}
+}
+
+func runCreate(ctx context.Context, name, specSource, serviceConfig, output, specFormat string) error {
+	return runCreateWithGenerator(ctx, name, specSource, serviceConfig, output, specFormat, &Generate{})
+}
+
+func runCreateWithGenerator(ctx context.Context, name, specSource, serviceConfig, output, specFormat string, gen Generator) error {
+	cfg, err := yaml.Read[config.Config](librarianConfigPath)
+	if err != nil {
+		return fmt.Errorf("%w: %v", errNoYaml, err)
+	}
+	switch cfg.Language {
+	case "rust":
+		for _, lib := range cfg.Libraries {
+			if lib.Name == name {
+				return gen.Run(ctx, false, name)
+			}
+		}
+
+		if serviceConfig == "" || specSource == "" {
```

Comment:
For veneer libraries this condition may be true as the service config and source are in rust modules.